### PR TITLE
chore(ESSNTL-4076): Upgrade react-router-dom to version 6

### PR DESCRIPTION
**Pull request by mkholjuraev**
Resolves: https://issues.redhat.com/browse/ESSNTL-4076

Upgrades react-router to v6 and refactors all class components to function components

### CODE REVIEW

File: src/components/InventoryTable/EntityTable.js [hunk 1]
```diff
@@ -71,7 +70,7 @@ const EntityTable = ({
   );
 
   const defaultRowClick = (_event, key) => {
-    history.push(
+    navigate(
```
Comment (gkarat) [inline src/components/InventoryTable/EntityTable.js:73]: I think here we can freely use `navigate` from react-router and just replace with `navigate(key)` (using relative path).

File: src/components/InventoryDetail/helpers.js [hunk 2]
```diff
@@ -12,7 +12,8 @@ export const redirectToInventoryList = (id, onBackToListClick) => {
       document.referrer !==
         `${document.location.origin}${document.location.pathname}`
     ) {
-      history.back();
+      console.log('hello');
```
Comment (gkarat) [inline src/components/InventoryDetail/helpers.js:15]:
```suggestion
```

File: src/Utilities/TestingUtilities.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+export const mountWithRouter = (Component, initialEntriest) => {
+  const wrapper = mount(
+    <MemoryRouter initialEntriest={initialEntriest}>{Component}</MemoryRouter>
```
Comment (gkarat) [inline src/Utilities/TestingUtilities.js:5]:
```suggestion
    <MemoryRouter initialEntries={initialEntries}>{Component}</MemoryRouter>
```

File: src/Utilities/TestingUtilities.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+export const mountWithRouter = (Component, initialEntriest) => {
```
Comment (gkarat) [inline src/Utilities/TestingUtilities.js:3]:
```suggestion
export const mountWithRouter = (Component, initialEntries) => {
```

**Review by gkarat — CHANGES_REQUESTED** [review]

File: src/Routes.js [hunk 5]
```diff
@@ -43,86 +43,52 @@ export const Routes = () => {
...
-            exact
-            path={routes.manageEdgeInventoryUrlName}
-            component={InventoryTable}
-            rootClass="inventory"
-          />
-          <Redirect path="*" to="/" />
-        </Switch>
-      )}
+
+  let element = useRoutes([
```
Comment (gkarat) [inline src/Routes.js:47]: I think we miss redirect here: see the old `<Redirect path="*" to="/" />`

**Comment by mkholjuraev** [comment]
@gkarat why there is a rage label?

**Comment by gkarat** [comment]
Haha, it just reflects my feelings working on the router upgrade 😄

**Review by gkarat — APPROVED** [review]
LGTM! The last issue was resolved, and I can see the app working fine locally. Let's push it next 🚀

**Pull request merged**